### PR TITLE
Run all Jest tests with SWC

### PR DESCRIPTION
**Pull request by Mrtenz**
Instead of `ts-jest`, we now use `@swc/jest` for running tests, which is much faster than `ts-jest`.

### CODE REVIEW

**Comment by Mrtenz** [comment]
@SocketSecurity ignore @jest/create-cache-key-function@27.5.1
@SocketSecurity ignore @swc/jest@0.2.26
@SocketSecurity ignore @jest/types@27.5.1
@SocketSecurity ignore @types/yargs-parser@21.0.0

File: packages/create-snap/src/cmds/init/init.test.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { initCommand } from '.';
 import * as initHandlerModule from './initHandler';
 
+jest.mock('./initHandler');
```
[the remark this answers is not in the document]
Comment (Mrtenz) [reply packages/create-snap/src/cmds/init/init.test.ts:4]: From my understanding: `jest.spyOn` replaces exports, which is not allowed in ESM world. Mocking it lets us override exports again.

File: packages/snaps-controllers/coverage.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "branches": 89.8,
-  "functions": 94.78,
-  "lines": 96.71,
-  "statements": 96.62
+  "functions": 95.28,
```
[the remark this answers is not in the document]
Comment (Mrtenz) [reply packages/snaps-controllers/coverage.json:3]: The build output is slightly different between TSC and SWC, so coverage is slightly different too.

File: packages/create-snap/src/cmds/init/init.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (FrederikBolding) [reply packages/create-snap/src/cmds/init/init.test.ts:4]: So we need `jest.mock` whereever we are trying to use ``jest.spyOn`?

[the remark this answers is not in the document]
Comment (Mrtenz) [reply packages/create-snap/src/cmds/init/init.test.ts:4]: Yep.

[the remark this answers is not in the document]
Comment (FrederikBolding) [reply packages/create-snap/src/cmds/init/init.test.ts:4]: We might need to document that somewhere? It doesn't feel obvious to me at least 😅

**Review by FrederikBolding — APPROVED** [review]

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**